.github/workflows/linux.yml

```diff
@@ -106,29 +106,33 @@ jobs:
 # TODO
 #  clang6_py36_nompi_h5_ad1_ad2_libcpp
 
-  clang10_py38_nompi_h5_ad1_ad2_libcpp:
+  clang12_py38_nompi_h5_ad1_ad2:
     runs-on: ubuntu-20.04
     if: github.event.pull_request.draft == false
     steps:
     - uses: actions/checkout@v2
+    - name: Spack Cache
+      uses: actions/cache@v2
+      with: {path: /opt/spack, key: clang12_py38_nompi_h5_ad1_ad2 }
     - name: Install
       run: |
         sudo apt-get update
-        sudo apt-get install clang-10 libc++-dev libc++abi-dev libhdf5-dev python3
-        python3 -m pip install -U pip
-        python3 -m pip install -U numpy
-#       TODO ADIOS1 ADIOS2
+        sudo apt-get install clang-12 libhdf5-dev python3 python3-pip python3-numpy python3-pandas
+        sudo .github/workflows/dependencies/install_spack
     - name: Build
-      env: {CC: clang-10, CXX: clang++-10, CXXFLAGS: -Werror -Wno-deprecated-declarations}
-      # TODO
-      # , CXXFLAGS: -nostdinc++ -I/usr/include/c++/10 -I/usr/include/x86_64-linux-gnu/c++/10/, LDFLAGS: -stdlib=libc++ -lc++ -lc++abi}
+      env: {CC: clang-12, CXX: clang++-12, CXXFLAGS: -Werror -Wno-deprecated-declarations}
       run: |
+        eval $(spack env activate --sh .github/ci/spack-envs/clang12_py38_nompi_h5_ad1_ad2/)
+        spack install
+
         share/openPMD/download_samples.sh build
         chmod u-w build/samples/git-sample/*.h5
         cmake -S . -B build \
           -DopenPMD_USE_PYTHON=ON \
           -DopenPMD_USE_MPI=OFF   \
           -DopenPMD_USE_HDF5=ON   \
+          -DopenPMD_USE_ADIOS1=ON \
```

Comment:
Weird problem:
```
/home/runner/work/openPMD-api/openPMD-api/.github/ci/spack-envs/clang12_py38_nompi_h5_ad1_ad2/.spack-env/view/include/adios_mpi.h:16:13: fatal error: 'mpi.h' file not found
#   include "mpi.h"
            ^~~~~~~
```
ADIOS1 should have an MPI mock in place, but it seems to be not found here.

Thus:
```suggestion
          -DopenPMD_USE_ADIOS1=OFF \
```